packages/mongodb-chatbot-eval/src/withConfig.ts

```diff
@@ -60,23 +60,26 @@ export const loadConfig = async ({
 };
 
 export const withConfig = async <T>(
-  action: (config: ResolvedConfig, args: T) => Promise<void>,
+  action: (config: EvalConfig, args: T) => Promise<void>,
   args: LoadConfigArgs & T
 ) => {
   const config = await loadConfig(args);
-  const [resolvedConfig, cleanup] = await resolveConfig(config);
   try {
-    return await action(resolvedConfig, args);
+    return await action(config, args);
+  } catch (error) {
+    logger.error(`Action failed: ${(error as Error).message}`);
+    throw error;
   } finally {
+    // Close all closable resources
     await Promise.all(
-      cleanup.map(async (close) => {
-        try {
-          await close();
-        } catch (error) {
-          logger.error(`Cleanup failed: ${(error as Error).message}`);
-        }
-      })
+      Object.values(config)
+        .filter((v) => v.close !== undefined)
+        .map((v) => v.close as () => Promise<void>)
     );
+    // Run afterAll if it exists
+    config.afterAll && (await config?.afterAll());
```

Comment:
```suggestion
    await config.afterAll?.();
```

`config` will always exist + we can use optional chaining instead of checking for `afterAll` separately